### PR TITLE
Fix manual instrumentation serialization for async tracks

### PR DESCRIPTION
**Pull request by pierricgimmig**
In the case of async time slices, the start and stop events are their
own individual timers and are already serialized on their initial thread
tracks. Make sure we don't serialize the async timers twice.

This change fixes a bug in the serialization of async timers that would
cause duplication of time slices. The other manual instrumentation
data was already properly serialized (scopes and graphs).

Fixes b/165078963

Test: grab a capture of OrbitTest which uses the manual instrumentation
API, save and reload.

### CODE REVIEW

**Review by florian-kuebler — APPROVED** [review]
Thanks Pierric for the change. It makes sense, even though this feels a bit hacky. However, I do not have a better solution in my mind. So LGTM.

File: OrbitGl/AsyncTrack.cpp [hunk 1]
```diff
@@ -51,6 +51,13 @@ void AsyncTrack::UpdateBoxHeight() {
   }
 }
 
+std::vector<std::shared_ptr<TimerChain>> AsyncTrack::GetAllChains() {
```
Comment (ronaldfw) [inline OrbitGl/AsyncTrack.cpp:54]: There's a similar issue with the frame tracks, which also is using data that is already present in other tracks. Do we need a clear concept of "derived tracks" that are never serialized, but are computed on load?

Comment (pierricgimmig) [reply OrbitGl/AsyncTrack.cpp]: I added an explicit GetAllSerializableChains() method that the tracks can override, I think the solution is cleaner now. Essentially, a track now has control on exactly what timers should be serialized.

**Comment by pierricgimmig** [comment]
> Thanks Pierric for the change. It makes sense, even though this feels a bit hacky.

:-) Yeah... I have to agree that it was a little bit hacky. I cleaned up the code by adding an explicit method to get serializable timers.

File: OrbitGl/AsyncTrack.cpp (hunk 1 above)
Comment (ronaldfw) [reply OrbitGl/AsyncTrack.cpp]: Great, I was going to suggest the same!

**Review by ronaldfw — APPROVED** [review]
LGTM!

File: OrbitGl/AsyncTrack.cpp (hunk 1 above)
Comment (florian-kuebler) [reply OrbitGl/AsyncTrack.cpp]: Yep, that makes it more clear. Thanks!

**Review by florian-kuebler — APPROVED** [review]
LGTM, thanks!

**Pull request merged**